chrome/content/zotero/xpcom/openurl.js

```diff
@@ -203,25 +203,33 @@ Zotero.OpenURL = new function() {
 			}
 		}
 		
-		if(item.creators && item.creators.length) {
-			// encode first author as first and last
-			var firstCreator = item.creators[0];
-			if(item.itemType == "patent") {
-				_mapTag(firstCreator.firstName, "invfirst");
-				_mapTag(firstCreator.lastName, "invlast");
-			} else {
-				if(firstCreator.isInstitution) {
-					_mapTag(firstCreator.lastName, "aucorp");
-				} else {
-					_mapTag(firstCreator.firstName, "aufirst");
-					_mapTag(firstCreator.lastName, "aulast");
+		var first = true;
+		for(var i=0; i<item.creators.length; i++) {
+			var creator = item.creators[i];
+			// only export authors
+			// other creator types cannot correctly export to COinS
+			if(creator.creatorType == "author") {
```

Comment:
Primary creator is not always "author". Use something like
```
var itemTypeID = Zotero.ItemTypes.getID(item.itemType);
var primaryCreator = Zotero.CreatorTypes.getName(Zotero.CreatorTypes.getPrimaryIDForType(itemTypeID));
...
if (creator.creatorType == primaryCreator) {
...
```
(though I haven't tested this)